src/server/game/Entities/Creature/CreatureData.h

```diff
@@ -355,33 +356,23 @@ typedef std::unordered_map<uint8, EquipmentInfo> EquipmentInfoContainerInternal;
 typedef std::unordered_map<uint32, EquipmentInfoContainerInternal> EquipmentInfoContainer;
 
 // from `creature` table
-struct CreatureData
+struct CreatureData : public SpawnData
 {
-    CreatureData() = default;
+    CreatureData() : SpawnData(SPAWN_TYPE_CREATURE) { }
     uint32 id1{0};                                             // entry in creature_template
```

Comment:
TC has completely removed the id columns. I don't know if 3 ids are useful for AC. Need feedback on this.